gen/generate.go

```diff
@@ -253,10 +259,16 @@ func generateModule(
 		return "", nil, err
 	}
 
+	// converts package name from ab-def to ab_def for golang code generation
+	packageRelPath, err := i.RelativePackage(m.ThriftPath)
+	if err != nil {
+		return "", nil, err
+	}
+	normalizedPackageName := filepath.Base(packageRelPath)
```

Comment:
This seems suspect. I think we want the original package path to remain the same, and to normalize here,

    packageName := normalizeRelativePath(filepath.Base(packageRelPath))

`RelativePackage` is supposed to be the import path to the generated code relative to the output directory. You can see that in the `Package` method implementation. It's basically a `join(ImportPrefix, RelativePackage)`.

So if you had ThriftRoot `foo/`, import prefix `example.com/gen/`, and a Thrift file at `foo/bar/baz-qux.thrift`, we want, `RelativePackage = "bar/baz-qux"`, which gives us import path `example.com/gen/bar/baz-qux`, and we want package name `baz_qux`.

Reply:
done.